lib/Config.js

```diff
@@ -1749,6 +1749,18 @@ class Config extends EventEmitter {
          */
         this.bypassMaxPutObjectSize = process.env.BYPASS_MAX_PUT_OBJECT_SIZE === 'true'
             || config.bypassMaxPutObjectSize || false;
+
+        /**
+         * S3C-10370: Before 9.5.1, there was no limit on the key length.
+         * The current limit is 915, provide a way to ignore (set to 0) or override the limit
+         */
+        this.overrideKeyLimit = process.env.OVERRIDE_KEY_LIMIT || config.overrideKeyLimit || null;
+        if (this.overrideKeyLimit !== null) {
+            this.overrideKeyLimit = parseInt(this.overrideKeyLimit, 10);
+            assert(Number.isInteger(this.overrideKeyLimit) && this.overrideKeyLimit >= 0,
+                'bad config: overrideKeyLimit must be a positive integer');
+        }
```

Comment:
For clarity, instead of overwriting the same attribute, it's best to use a const variable:

```suggestion
        const overrideKeyLimitStr = process.env.OVERRIDE_KEY_LIMIT || config.overrideKeyLimit || null;
        if (overrideKeyLimitStr !== null) {
            this.overrideKeyLimit = parseInt(overrideKeyLimitStr, 10);
            assert(Number.isInteger(this.overrideKeyLimit) && this.overrideKeyLimit >= 0,
                'bad config: overrideKeyLimit must be a positive integer');
        } else {
            this.overrideKeyLimit = null;
        }
```

Reply:
There [61178cc](https://github.com/scality/cloudserver/pull/5942/commits/61178ccffb6fe678970c5428a69171bfbdeccc15)